python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -874,7 +879,14 @@ def __name__(self, value):
         setattr(self._fsproxy_slow, "__name__", value)
 
 
-def _fast_slow_function_call(func: Callable, /, *args, **kwargs) -> Any:
+def _fast_slow_function_call(
+    func: Callable,
+    /,
+    *args,
+    debug_mode: str = "mode.pandas_debugging",
+    assert_func: Callable = assert_eq,
```

Comment:
*issue*: this interleaves the arguments that will be passed to `func` with the arguments that control the behaviour of `_fast_slow_function_call` itself.

Moreover, it means that `func` may not take the arguments `debug_mode` or `assert_func` by name.

Do we need to rejig this a bit so it's something like:

```
def _fast_slow_function_call(
    func: Callable,
    debug_mode: str | None = None,
    assert_func: Callable | None = None,
    args: Sequence[Any],
    kwargs: Mapping[Any, Any]
) -> Any:
```

And then the caller (in cudf.pandas) doesn't splat `args` and `kwargs`.

Additionally, it makes partially evaluating the function neater, we do:
```
fast_slow = partial(_fast_slow_function_call, debug_mode="mode.pandas_debugging", assert_func=assert_eq)
```
Just in the pandas-specific wrapper code.